### PR TITLE
refactor(types): split up widget.ts

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

the types in Widget.ts aren't all directly related to widgets, which makes it very hard to read

**Result**

more smaller files

### CODE REVIEW

File: src/types/widget.ts [hunk 1]
```diff
@@ -473,7 +172,9 @@ export type Widget<
    */
   getWidgetSearchParameters?(
     state: SearchParameters,
-    widgetSearchParametersOptions: WidgetSearchParametersOptions
+    widgetSearchParametersOptions: {
```
Comment (eunjae-lee) [inline src/types/widget.ts:175]: Any reason you removed `WidgetSearchParametersOptions` type and access it via `Parameters` later?

**Review by eunjae-lee — APPROVED** [review]
Awesome. Thanks for cleaning it up!

File: src/types/widget.ts (hunk 1 above)
Comment (Haroenv) [reply src/types/widget.ts:175]: the type later (PR to come) depends on the generic passed to Widget, so the WidgetSearchParametersOptions would need the generic too. I might revisit this part once its migrated fully!

**Review by francoischalifour — COMMENTED** [review]
A convention that I see more often is to name the type files like the type.

Example: `UiState.ts`

**Comment by Haroenv** [comment]
What if it exports more than one item @francoischalifour? for example `instantsearch` exports `InstantSearch` & `InstantSearchOptions` etc. Case-only name changes are also a little tricky on git

**Comment by francoischalifour** [comment]
@Haroenv We could either export `InstantSearchOptions` from `InstantSearch.ts` (since it's closely related), or create another `InstantSearchOptions.ts` file.

**Comment by Haroenv** [comment]
In https://github.com/algolia/instantsearch.js/pull/4734 I'm adding different exports in most of those files (IndexUiState / UiState in ui-state for example). Do you think those should be in their own file or something IndexUiState should be exported from UiState.ts?

**Comment by francoischalifour** [comment]
I think it's OK for `IndexUiState` to be exported from `UiState.ts`.

**Review by tkrugg — APPROVED** [review]

**Pull request merged**